scripts/dataset_processing/tts/preprocess_text.py

```diff
@@ -0,0 +1,129 @@
+    --normalizer_config_path="<nemo_root_path>/examples/tts/conf/text/normalizer_en.yaml" \
+    --lower_case=True \
+    --num_workers=1
+"""
+
+import argparse
+from pathlib import Path
+
+from hydra.utils import instantiate
+from joblib import Parallel, delayed
+from nemo_text_processing.text_normalization.normalize import Normalizer
+from omegaconf import OmegaConf
+from tqdm import tqdm
+
+from nemo.collections.asr.parts.utils.manifest_utils import read_manifest, write_manifest
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter, description="Process and normalize text data.",
+    )
+    parser.add_argument(
+        "--input_manifest", required=True, type=Path, help="Path to input training manifest.",
+    )
+    parser.add_argument(
+        "--output_manifest", required=True, type=Path, help="Path to output training manifest with processed text.",
+    )
+    parser.add_argument(
+        "--overwrite", default=False, type=bool, help="Whether to overwrite the output manifest file if it exists.",
+    )
+    parser.add_argument(
+        "--lower_case", default=False, type=bool, help="Whether to convert the final text to lower case.",
+    )
+    parser.add_argument(
+        "--normalizer_config_path",
+        required=False,
+        type=Path,
+        help="Path to config file for nemo_text_processing.text_normalization.normalize.Normalizer.",
+    )
+    parser.add_argument(
+        "--num_workers", default=1, type=int, help="Number of parallel threads to use. If -1 all CPUs are used."
+    )
+    parser.add_argument(
+        "--max_entries", default=0, type=int, help="If provided, maximum number of entries in the manifest to process."
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def _process_entry(entry: dict, normalizer: Normalizer, lower_case: bool, lower_case_norm: bool) -> dict:
+    text = entry["text"]
+
+    if normalizer is not None:
+        if lower_case_norm:
+            text = text.lower()
+        text = normalizer.normalize(text, punct_pre_process=True, punct_post_process=True)
+
+    if lower_case:
+        text = text.lower()
+
+    entry["normalized_text"] = text
+
+    return entry
+
+
+def main():
+    args = get_args()
+
+    input_manifest_path = args.input_manifest
+    output_manifest_path = args.output_manifest
+    lower_case = args.lower_case
+    num_workers = args.num_workers
+    max_entries = args.max_entries
+    overwrite = args.overwrite
+
+    if output_manifest_path.exists():
+        if overwrite:
+            print(f"Will overwrite existing manifest path: {output_manifest_path}")
+        else:
+            raise ValueError(f"Manifest path already exists: {output_manifest_path}")
+
+    if args.normalizer_config_path:
+        normalizer_config = OmegaConf.load(args.normalizer_config_path)
+        lower_case_norm = normalizer_config.input_case == "lower_cased"
```

Comment:
Maybe we could keep it as field in `nemo_text_processing`? Feels wrong to drag it around.

Reply:
I mean, make it a field in `Normalizer` class. 

https://github.com/NVIDIA/NeMo-text-processing/blob/9afbaf20c569ed41567988956e0e90dc913dd516/nemo_text_processing/text_normalization/normalize.py#L97

Put a `self.input_case = input_case` clause in the initializer

Reply:
Makes sense to me. I will set up the `nemo_text_processing` repo when I have time.

Reply:
Changed.